### ISSUE TITLE
Results order of pre-trained story generation model

### ISSUE THREAD

**Issue by cairoHy** [pytorch/fairseq#1369]
Hi, I download the 150 word generated stories from `https://dl.fbaipublicfiles.com/fairseq/data/fusion_stories.txt`, but the results file contains 15137 lines(one sample each line) while writingPrompts test file: `test.wp_source` contains 15138 lines, and also it seems the order of the results is not the same as the test set prompt file.
Wish if I made some mistake or could you get some help~
@huihuifan

**Comment by huihuifan** [comment]
The order is ordered by input length, as generate.py does some sorting to try to batch the generations for efficiency (the codebase started for translation, and sentences of similar input lengths tend to have similar output lengths, so if you sort by input, there is less waiting time for very long sentences in the batch). However, it is possible some code changes since I produced this file almost 2 years ago means it could be hard to recover the order. To remedy this, I have uploaded another file that corresponds to the prompt order: 

https://dl.fbaipublicfiles.com/fairseq/data/fusion_prompts.txt  which has 15137 lines

and will update the stories readme to reference this. Thanks for pointing this out! Note that we chose stories for evaluation where the prompt was not an unk - at the time, we didn't use BPE, model a very large vocabulary, or use pretraining, so there are unk. 

For the 1 missing story, I apologize but I am not sure why it is missing - this was the exact file we used for the evaluation.
